### ISSUE TITLE
SOCKSv4a tests use a real DNS resolver, causing intermittent test failures

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4275]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)| @jesstess reported|
|-|-|
|Trac ID|trac#4275|
|Type|release blocker: regression|
|Created|2010-02-12 18:21:19Z|
|Branch|https://github.com/twisted/twisted/tree/socks-resolver-regression-4275|

The Bind test should have used FakeResolverReactor as in the Connect case.

While we're at it, the Bind test case should have a SOCKSv4a failure test like in Connect.

<details><summary>Searchable metadata</summary>

```
trac-id__4275 4275
type__release_blocker__regression release blocker: regression
reporter__jesstess jesstess
priority__highest highest
milestone__ 
branch__branches_socks_resolver_regression_4275 branches/socks-resolver-regression-4275
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1265998879000000 1265998879000000
changetime__1266597169000000 1266597169000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_1"></a>|@jesstess commented|
|-|-|

(In [28181]) Branching to 'socks-resolver-regression-4275'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_2"></a>|@jesstess commented|
|-|-|

(In [28182]) All SOCKSv4a tests should use a dummy reactor with a deterministic resolve method.

refs [#4275](https://github.com/twisted/twisted/issues/4275)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @jesstess|
|-|-|

Just one issue:

  1. `test_socks4aFailedResolution` starts with a comment that incorrectly talks about "localhost"

Please fix that and merge.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess commented|
|-|-|

(In [28243]) Fix comment about 'localhost' being resolved when it's actually
'failinghost'.

refs [#4275](https://github.com/twisted/twisted/issues/4275)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess set status to `closed`|
|-|-|

(In [28245]) Merge socks-resolver-regression-4275

Author: jesstess
Reviewer: exarkun
Fixes: [#4275](https://github.com/twisted/twisted/issues/4275)

Commits for [#3886](https://github.com/twisted/twisted/issues/3886) caused a regression in which the use of a real
resolver caused intermittent SOCKS test failures. All SOCKSv4a tests
now use a dummy reactor with a deterministic resolve method to avoid
this problem.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
